components/camel-as2/camel-as2-api/src/main/java/org/apache/camel/component/as2/api/entity/AS2MessageDispositionNotificationEntity.java

```diff
@@ -0,0 +1,262 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.as2.api.entity;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.camel.component.as2.api.AS2Charset;
+import org.apache.camel.component.as2.api.AS2Header;
+import org.apache.camel.component.as2.api.AS2MimeType;
+import org.apache.camel.component.as2.api.CanonicalOutputStream;
+import org.apache.camel.component.as2.api.util.HttpMessageUtils;
+import org.apache.camel.component.as2.api.util.MicUtils;
+import org.apache.camel.component.as2.api.util.MicUtils.ReceivedContentMic;
+import org.apache.http.Header;
+import org.apache.http.HeaderIterator;
+import org.apache.http.HttpEntityEnclosingRequest;
+import org.apache.http.HttpException;
+import org.apache.http.HttpResponse;
+import org.apache.http.entity.ContentType;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.Args;
+
+public class AS2MessageDispositionNotificationEntity extends MimeEntity {
+
+    private static final String ADDRESS_TYPE_PREFIX = "rfc822;";
+    private static final String MTA_NAME_TYPE_PREFIX = "dns;";
+    private static final String REPORTING_UA = "Reporting-UA";
+    private static final String MDN_GATEWAY = "MDN-Gateway";
+    private static final String FINAL_RECIPIENT = "Final-Recipient";
+    private static final String ORIGINAL_MESSAGE_ID = "Original-Message-ID";
+    private static final String AS2_DISPOSITION = "Disposition";
+    private static final String FAILURE = "Failure";
+    private static final String ERROR = "Error";
+    private static final String WARNING = "Warning";
+    private static final String RECEIVED_CONTENT_MIC = "Received-content-MIC";
+
+    private String reportingUA;
+    // TODO determine if we need to support this field.
```

Comment:
is it a leftover? or there is a specific task for it?

Reply:
No. I put this comment in to remind myself to reevaluate whether the MDN gateway field feature is ever being used once the component gets into the field.